src/server/scripts/Outland/TempestKeep/arcatraz/boss_harbinger_skyriss.cpp

```diff
@@ -181,107 +182,145 @@ struct boss_harbinger_skyriss : public BossAI
         }
+
         if (!UpdateVictim())
             return;
 
-        if (!IsImage66 && !HealthAbovePct(66))
-        {
-            DoSplit(66);
-            IsImage66 = true;
-        }
-        if (!IsImage33 && !HealthAbovePct(33))
-        {
-            DoSplit(33);
-            IsImage33 = true;
-        }
+        events.Update(diff);
 
-        if (MindRend_Timer <= diff)
-        {
-            if (Unit* target = SelectTarget(SelectTargetMethod::Random, 1))
-                DoCast(target, SPELL_MIND_REND);
-            else
-                DoCastVictim(SPELL_MIND_REND);
-
-            MindRend_Timer = 8000;
-        }
-        else
-            MindRend_Timer -=diff;
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
 
-        if (Fear_Timer <= diff)
+        while (uint32 eventId = events.ExecuteEvent())
         {
-            if (me->IsNonMeleeSpellCast(false))
-                return;
+            switch (eventId)
+            {
+                case EVENT_MIND_REND:
+                    if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
+                        DoCast(target, SPELL_MIND_REND);
+                    events.Repeat(8s, 12s);
+                    break;
+                case EVENT_FEAR:
+                    if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
+                        DoCast(target, SPELL_FEAR);
+                    events.Repeat(25s, 35s);
+                    break;
+                case EVENT_DOMINATION:
+                    if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
+                        DoCast(target, SPELL_DOMINATION);
+                    events.Repeat(30s, 40s);
+                    break;
+                case EVENT_MANA_BURN:
+                    if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
+                        DoCast(target, SPELL_MANA_BURN);
+                    events.Repeat(15s, 30s);
+                    break;
+                case EVENT_SUMMON_66:
+                    Talk(SAY_IMAGE);
+                    DoCastSelf(SPELL_SUMMON_66_ILLUSION);
+                    DoCastSelf(SPELL_BLINK_VISUAL);
+                    break;
+                case EVENT_SUMMON_33:
+                    Talk(SAY_IMAGE);
+                    DoCastSelf(SPELL_SUMMON_33_ILLUSION);
+                    DoCastSelf(SPELL_BLINK_VISUAL);
+                    break;
+                default:
+                    break;
+            }
 
-            Talk(SAY_FEAR);
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+        }
 
-            if (Unit* target = SelectTarget(SelectTargetMethod::Random, 1))
-                DoCast(target, SPELL_FEAR);
-            else
-                DoCastVictim(SPELL_FEAR);
+        DoMeleeAttackIfReady();
+    }
 
-            Fear_Timer = 25000;
-        }
-        else
-            Fear_Timer -=diff;
+private:
+    bool _intro;
+    bool _image33;
```

Comment:
i would merge `_image33` and `_image66` into an `uint8` and then use value 1 and 2 respectively

Reply:
Show me an example please

Reply:
declare it as `uint8 _imagePhase`

create an enum for the phases
```c
enum SkyrissImagePhases : uint8
{
    IMAGE_PHASE_NONE = 0,
    IMAGE_PHASE_66,
    IMAGE_PHASE_33,
}
```

in constructor initialize it as `_imagePhase(IMAGE_PHASE_NONE)`, do the same in `Reset()`

and then

```c
        if (_imagePhase < IMAGE_PHASE_66 && me->HealthBelowPctDamaged(66, damage))
        {
            _imagePhase++;
            events.ScheduleEvent(EVENT_SUMMON_66, 0s);
        }
        if (_imagePhase < IMAGE_PHASE_33 && me->HealthBelowPctDamaged(33, damage))
        {
            _imagePhase++;
            events.ScheduleEvent(EVENT_SUMMON_33, 0s);
        }
```

Reply:
To be honest that looks overcomplicated than simple bool